src/controllers/api/users.js

```diff
@@ -0,0 +1,155 @@
+const {User, Thought} = require("../../models");
+
+const getUsers = async (req, res) => {
+	try {
+		const users = await User.find({});
+
+		if (!users.length) {
+			return res.status(404).json({
+				success: false,
+				message: "No user's exist in database",
+			});
+		} else {
+			return res.status(200).json({
+				success: true,
+				data: users,
+			});
+		}
+	} catch (error) {
+		return res.status(500).json({
+			success: false,
+			message: "Failed to get users",
+			error: error.message,
+		});
+	}
+};
+
+const getUserById = async (req, res) => {
+	try {
+		const userId = req.params.id;
+
+		const user = await User.findById(userId)
+			.populate("thoughts")
+			.populate("friends");
+
+		return res.status(200).json({
+			success: true,
+			data: user,
+		});
+	} catch (error) {
+		return res.status(500).json({
+			success: false,
+			message: "Failed to get user",
+			error: error.message,
+		});
+	}
+};
+
+const createUser = async (req, res) => {
+	try {
+		const user = req.body;
+
+		if (!user.username) {
+			return res.status(400).json({
+				success: false,
+				message: "No username supplied",
+			});
+		} else if (!user.email) {
+			return res.status(400).json({
+				success: false,
+				message: "No email supplied",
+			});
+		} else {
+			const createdUser = await User.updateOne(
+				user,
+				{$set: {username: user.username, email: user.email}},
+				{upsert: true}
+			);
+
+			if (createdUser.matchedCount) {
+				return res.status(409).json({
+					success: false,
+					message: "User already exists",
+				});
+			} else {
+				return res.status(200).json({
+					success: true,
+					data: createdUser,
+				});
+			}
+		}
+	} catch (error) {
+		return res.status(500).json({
+			success: false,
+			message: "Failed to create User",
+			error: error.message,
+		});
+	}
+};
+
+const updateUserById = async (req, res) => {
+	try {
+		const {username, email} = req.body;
+
+		if (!username) {
```

Comment:
What if someone only wants to update either username or email? How can we allow this?